libvast/vast/concepts.hpp

```diff
@@ -21,6 +21,12 @@ concept SameHelper = std::is_same_v<T, U>;
 template <class T, class U>
 concept same_as = SameHelper<T, U> && SameHelper<U, T>;
 
+template <class T, class U>
+concept sameish = same_as<std::decay_t<T>, std::decay_t<U>>;
```

Comment:
This needs to be defined differently: https://stackoverflow.com/questions/58509147/why-does-same-as-concept-check-type-equality-twice

Reply:
Mind elaborating? We're routing through the `same_as` concept defined right above which goes through `SameHelper` so we check that `T == U` and `U == T`.

Reply:
You are absolutely correct—must've confused SameHelper and same_as yesterday when glancing at the code.